tasks.py

```diff
@@ -758,20 +793,26 @@ def ruff(context, action=None, target=None, fix=False, output_format="concise"):
     if not target:
         target = ["."]
 
+    exit_code = 0
+
     if "format" in action:
         command = "ruff format "
         if not fix:
             command += "--check "
         command += " ".join(target)
-        run_command(context, command, warn=True)
+        if not run_command(context, command, warn=True):
+            exit_code = 1
 
     if "lint" in action:
         command = "ruff check "
         if fix:
             command += "--fix "
         command += f"--output-format {output_format} "
         command += " ".join(target)
-        run_command(context, command, warn=True)
+        if not run_command(context, command, warn=True):
+            exit_code = 1
+
+    raise Exit(code=exit_code)
```

Comment:
From my testing these should be behind a check of the exit_code as it seems that even a code of 0 causes it to exit causing the tests when run locally to stop and not continue to the next one.

Reply:
As discussed, this should be fixed in the next version of the cookie being pushed out.